nodedb.go

```diff
@@ -125,11 +125,18 @@ func (ndb *nodeDB) GetNode(hash []byte) (*Node, error) {
 
 	// Check the cache.
 	if elem, ok := ndb.nodeCache[string(hash)]; ok {
+		if ndb.opts.Stat != nil {
```

Comment:
Could you please also add a couple of unit tests with and without the `nil` `Stat`. 

Also, asserting that the counters get increased would be great.

Reply:
Done